test/inductor/test_external_callables.py

```diff
@@ -0,0 +1,69 @@
+# Owner(s): ["module: inductor"]
+import torch
+
+from torch._inductor import config
+
+from torch._inductor.test_case import run_tests, TestCase
+
+
+class MatMulModule(torch.nn.Module):
+    def __init__(self):
+        super()
+        self.matrix = torch.nn.Parameter(torch.eye(128, 128) * 2, requires_grad=True)
+
+    def forward(self, x):
+        return torch.matmul(x, self.matrix)
+
+
+# torch.add performs better than torch.mm and got choosed during tuning
+def matmul_cpu(a: torch.Tensor, b: torch.Tensor, out: torch.Tensor) -> None:
+    torch.add(a, b, out=out)
+
+
+def matmul_cuda(a: torch.Tensor, b: torch.Tensor, out: torch.Tensor) -> None:
+    torch.add(a, b, out=out)
+
+
+class TestInductorExternalCallable(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls._saved_config = config.save_config()
+
+    def tearDown(self):
+        super().tearDown()
+        config.load_config(self._saved_config)
+
+    def test_matmul_cpu(self):
+        # 2I + 2I == (2I)(2I)
+        x = torch.eye(128, 128) * 2
+        opt_fn = torch.compile(
+            MatMulModule(),
+            options={"max_autotune": True, "external_matmul": [matmul_cpu]},
+        )
+        opt_fn_golden = torch.compile(MatMulModule(), options={"max_autotune": True})
+        torch.testing.assert_close(
+            opt_fn(x),
+            opt_fn_golden(x),
+            msg=f"torch.compile(..., external_matmul = {matmul_cpu}) failed",
+        )
+
+    def test_matmul_cuda(self):
```

Comment:
May need to skip this test when cuda isn't available, check what CI things.

Reply:
I added skip if no cuda is found